packages/database/src/models/ImagingRequestArea/hooks.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  INVOICE_ITEMS_CATEGORIES,
+  INVOICEABLE_IMAGING_REQUEST_STATUSES,
+  REFERENCE_TYPES,
+} from '@tamanu/constants';
+import type { ImagingRequestArea } from './ImagingRequestArea';
+import type { ImagingRequest } from 'models/ImagingRequest';
+
+const addItemAsAreaProduct = async (
+  instance: ImagingRequestArea,
+  imagingRequest: ImagingRequest,
+) => {
+  if (!imagingRequest.encounterId) {
+    return false;
+  }
+
+  if (!instance.areaId) {
+    return false;
+  }
+
+  const areaProduct = await instance.sequelize.models.InvoiceProduct.findOne({
+    where: {
+      category: INVOICE_ITEMS_CATEGORIES.IMAGING_AREA,
+      sourceRecordId: instance.areaId,
+    },
+  });
+
+  if (areaProduct) {
+    await instance.sequelize.models.Invoice.addItemToInvoice(
+      instance,
+      imagingRequest.encounterId,
+      areaProduct,
+      imagingRequest.requestedById,
+    );
+    return true;
+  }
+
+  return false;
+};
+
+const addItemAsRequestTypeProduct = async (
+  instance: ImagingRequestArea,
+  imagingRequest: ImagingRequest,
+) => {
+  if (!imagingRequest.encounterId) {
+    return false;
+  }
+
+  const requestType = await instance.sequelize.models.ReferenceData.findOne({
+    where: {
+      type: REFERENCE_TYPES.IMAGING_TYPE,
+      code: imagingRequest.imagingType,
+    },
+  });
+  if (!requestType) {
+    return false;
+  }
+
+  const requestTypeProduct = await instance.sequelize.models.InvoiceProduct.findOne({
+    where: {
+      category: INVOICE_ITEMS_CATEGORIES.IMAGING_TYPE,
+      sourceRecordId: requestType.id,
+    },
+  });
+  if (!requestTypeProduct) {
+    return;
+  }
+
+  const areaRefData = await instance.sequelize.models.ReferenceData.findByPk(instance.areaId);
+  // Add note of area for reference
+  await instance.sequelize.models.Invoice.addItemToInvoice(
+    instance,
+    imagingRequest.encounterId,
+    requestTypeProduct,
+    imagingRequest.requestedById,
+    areaRefData?.name,
+  );
+};
+
+const addToInvoice = async (instance: ImagingRequestArea) => {
+  const imagingRequest = await instance.sequelize.models.ImagingRequest.findByPk(
+    instance.imagingRequestId,
+  );
+  if (!imagingRequest || !imagingRequest.encounterId) {
+    return;
+  }
+
+  if (!INVOICEABLE_IMAGING_REQUEST_STATUSES.includes(imagingRequest.status)) {
+    return;
+  }
+
+  // Remove the existing imaging request item from the invoice as now we're tracking the items by their areas
+  await instance.sequelize.models.Invoice.removeItemFromInvoice(
```

Comment:
I could be mis-understanding something but I expected this to just remove the InvoiceItem for the ImagingRequest but it removes all the InvoiceItems for the Encounter. There could be multiple ImagingRequests on the Encounter including a mix of Imaging Requests with or without associated Imaging Requests Areas, right?